### DISCUSSION TITLE
NetworkChaos injection failed because the device could not be found

### DISCUSSION THREAD

**Discussion by tddhit** [chaos-mesh/chaos-mesh#3953] [Q&A] [answered]
## Bug Report

**What version of Kubernetes are you using?**
v1.18.4

**What version of Chaos Mesh are you using?**
v2.5.1

**What did you do? / Minimal Reproducible Example**
The code related to NetworkChaos is hard-coded to use eth0, while our container NIC is eth1, causing the injection to fail
![image](https://user-images.githubusercontent.com/8747237/221134552-9203f37f-fdb7-4e8d-9bf2-373e89e6acf2.png)
The SetTcs function in tc_server.go uses getAllInterfaces to get the actual device, which is missing in iptables_server.go

**Comment by STRRL** [comment, marked as answer]
You could use `NetworkChaos.spec.device` to claim the device name.

BTW, what's the CNI you were using?

**Reply by tddhit** [reply]
Got it, thank you. We are using Tencent Cloud's vpc-cni

**Reply by tddhit** [reply]
Is it more appropriate to get device through chaosdaemon getAllInterfaces
